File: Domain/WorkFlows/WorkFlowConfiguration.cs
using System.ComponentModel.DataAnnotations.Schema;
using BergerMsfaApi.Core;
using BergerMsfaApi.Domain.Organizations;
using BergerMsfaApi.Enumerations;

namespace BergerMsfaApi.Domain.WorkFlows
{

    public class WorkFlowConfiguration : AuditableEntity<int>
    {
        public int OrgRoleId { get; set; }

        public int MasterWorkFlowId { get; set; }

        [ForeignKey("MasterWorkFlowId")]
        public WorkFlow WorkFlow { get; set; }
        [ForeignKey("OrgRoleId")]
        public OrganizationRole OrganizationRole { get; set; }

        public ModeOfApproval ModeOfApproval { get; set; }
        public ApprovalStatus ApprovalStatus { get; set; }
        public ReceivedStatus ReceivedStatus { get; set; }
        public RejectedStatus RejectedStatus { get; set; }
        public NotificationStatus NotificationStatus { get; set; }

        public int sequence { get; set; }
    }
}

File: ClientApp/src/app/Shared/Entity/WorkFlows/workflowconfiguration.ts
import { Status } from "../../Enums/status";
import { YesNo } from "../../Enums/yesno";

export class WorkFlowConfiguration {
    public id: number;
    public orgRoleId: number;
    public orgRoleName: string;
    public masterWorkFlowId: number;
    public masterWorkFlowName: string;
    //public workFlow: WorkFlow;
    public modeOfApproval: number;
    public approvalStatus: number;
    public rejectedStatus: number;
    public notificationStatus: number;
    public status: number;
    public sequence : number;
    constructor() {
        this.id = 0;
        this.status = Status.Active;
    }
}